### ISSUE TITLE
Win10: CMD hangs on CTRL+C event after GET request

### ISSUE THREAD

**Issue by gimmx** [sc0tfree/updog#4]
Starting the server with "updog" allows stopping the server with CTRL+C. However, after navigation and subsequent console output, the terminal hangs when attempting to capture CTRL+C event. The delay seems to be coupled with some internal I/O block and does not properly listen for the stop signal.

**Comment by sc0tfree** [comment]
I have verified this on Windows 10. Seems like SIGINT stops sending only after a file download as it still works after just a GET request.

Can't seem to find out the right fix, but looks like there is an issue with werkzeug and signal libraries on Windows.

Anyone have ideas on this one?

https://github.com/sc0tfree/updog/blob/28a1ac16122fe520e29f7b21cffe85744771e216/updog/__main__.py#L174

**Comment by adeutscher** [comment]
This might be a bit apples and oranges, but I had a control-C problem with my SimpleHTTPServer+ThreadingMixIn-based script. Ctrl-C-ing out of the script while a download was in motion would hang until every client thread died down.

In order to solve this, I overrode the method responsible for serving content with a version that checked for an 'alive' attribute on the main server object. On a ctrl-C, 'alive' is set to `False` and the client file-serving loop breaks. This allows the client thread to die. https://github.com/adeutscher/core-tools/blob/93efb12e780964fd4079c79201e7966e2824327b/scripts/networking/http-servers/CoreHttpServer.py#L682

**Comment by sc0tfree** [comment]
Closing as stale (5 years old, no recent activity). If this is still an issue with v2.0, please open a new issue with current reproduction steps and environment details.
